### PR TITLE
docs: fix formatting of docstring with lists

### LINKED ISSUE #1345: Sphinx warning in generated client for google/cloud/gkebackup/v1
The generated client for google/cloud/gkebackup/v1 contains a docstring with a list which is not formatted correctly resulting in the following errors:

```
sphinx.errors.SphinxWarning: /usr/local/google/home/partheniou/git/python-gke-backup/google/cloud/gke_backup_v1/services/backup_for_gke/async_client.py:docstring of google.cloud.gke_backup_v1.services.backup_for_gke.async_client.BackupForGKEAsyncClient.create_backup:44:Block quote ends without a blank line; unexpected unindent.

Warning, treated as error:
/usr/local/google/home/partheniou/git/python-gke-backup/google/cloud/gke_backup_v1/services/backup_for_gke/async_client.py:docstring of google.cloud.gke_backup_v1.services.backup_for_gke.async_client.BackupForGKEAsyncClient.create_backup:44:Block quote ends without a blank line; unexpected unindent.
```

Source:
https://github.com/googleapis/googleapis-gen/blob/master/google/cloud/gkebackup/v1/gkebackup-v1-py/google/cloud/gke_backup_v1/services/backup_for_gke/async_client.py#L873

```
            backup_id (:class:`str`):
                The client-provided short name for
                the Backup resource. This name must:

                 - be between 1 and 63 characters long
                (inclusive)  - consist of only
                lower-case ASCII letters, numbers, and
                dashes  - start with a lower-case letter
                 - end with a lower-case letter or
                number
                 - be unique within the set of Backups
                in this BackupPlan
```

Marking this as P2 as I've added a workaround in the client library to allow the docs check to pass.

### PR DESCRIPTION
**Pull request by parthea**
Fixes https://github.com/googleapis/gapic-generator-python/issues/1345 🦕

The PR applies the following fixes for the function that converts documentation to RST format.

- Each item in a list should have a new line
- The new line after a colon in a list should be preserved
- When new lines wrap in a list, they should be indented.

All of the changes have corresponding tests to show the behaviour without the fixes.

### CODE REVIEW

File: gapic/utils/lines.py [hunk 1]
```diff
@@ -115,7 +123,9 @@ def wrap(text: str, width: int, *, offset: Optional[int] = None, indent: int = 0
         text='\n'.join([textwrap.fill(
             break_long_words=False,
             initial_indent=' ' * indent,
-            subsequent_indent=' ' * indent,
+            # ensure that subsequent lines for lists are indented 2 spaces
+            subsequent_indent=' ' * indent + \
```
Comment (dizcology) [inline gapic/utils/lines.py:127]: Maybe add spaces around `=`.

**Review by dizcology — APPROVED** [review]

File: gapic/utils/lines.py [hunk 2]
```diff
@@ -86,8 +86,12 @@ def wrap(text: str, width: int, *, offset: Optional[int] = None, indent: int = 0
                                 break_on_hyphens=False,
                                 )
         # Strip the first \n from the text so it is not misidentified as an
-        # intentionally short line below.
-        text = text.replace('\n', ' ', 1)
+        # intentionally short line below, except when the text contains `:`
+        # as the new line is required for lists.
+        if '\n' in text:
+            initial_text = text.split('\n')[0]
+            if ":" not in initial_text:
```
Comment (dizcology) [inline gapic/utils/lines.py:93]: Could `:` appear on text line 2, and the list starts from line 3 and on?

File: gapic/utils/lines.py (hunk 1 above)
Comment (parthea) [reply gapic/utils/lines.py:127]: I tried making this change but the style check failed
```
(py39) partheniou@partheniou-vm-3:~/git/gapic-generator-python$ find gapic tests -name "*.py" -not -path 'tests/**/goldens/*' | xargs autopep8 --diff --exit-code
--- original/gapic/utils/lines.py
+++ fixed/gapic/utils/lines.py
@@ -124,7 +124,7 @@
             break_long_words=False,
             initial_indent=' ' * indent,
             # ensure that subsequent lines for lists are indented 2 spaces
-            subsequent_indent = ' ' * indent + \
+            subsequent_indent=' ' * indent + \
             ('  ' if token.strip().startswith('-') else ''),
             text=token,
             width=width,
```

File: gapic/utils/lines.py (hunk 2 above)
Comment (parthea) [reply gapic/utils/lines.py:93]: yes, I added a test case in e08978a

File: gapic/utils/lines.py (hunk 1 above)
Comment (dizcology) [reply gapic/utils/lines.py:127]: I wonder if enclosing the right side in parentheses, or breaking this down into if-else statements would work.

**Review by dizcology — APPROVED** [review]

**Pull request merged**